crates/aptos-ledger/src/lib.rs

```diff
@@ -143,17 +143,86 @@ pub fn get_app_name() -> Result<String, AptosLedgerError> {
     }
 }
 
-/// Returns the public key of your Aptos account in Ledger device
+/// Returns the next available account in Ledger starting from start_index to end_index, inclusive
+/// Note: We only allow a range of 20 for performance purpose
+///
+/// # Arguments
+///
+/// * `start_index` - The index to start fetch for accounts
+/// * `end_index` - The last index to fetch the accounts
+pub fn fetch_batch_accounts(
+    start_index: u32,
+    end_index: u32,
+) -> Result<HashMap<String, AccountAddress>, AptosLedgerError> {
+    // Make sure start_index and end_index is valid
+    if start_index > end_index || end_index - start_index > 20 {
+        return Err(AptosLedgerError::UnexpectedError("Unexpected Error: Incorrect start and end index, make sure the range is less than or equal to 20".to_string(), None));
+    }
+
+    // Open connection to ledger
+    let transport = open_ledger_transport()?;
+
+    let mut accounts = HashMap::new();
+    for i in start_index..end_index {
+        let path = DERIVATIVE_PATH.replace("{index}", &i.to_string());
+        let cdata = serialize_bip32(&path);
+
+        match transport.exchange(&APDUCommand {
+            cla: CLA_APTOS,
+            ins: INS_GET_PUB_KEY,
+            p1: P1_NON_CONFIRM,
+            p2: P2_LAST,
+            data: cdata,
+        }) {
+            Ok(response) => {
+                // Got the response from ledger after user has confirmed on the ledger wallet
+                if response.retcode() == APDU_CODE_SUCCESS {
+                    // Extract the Public key from the response data
+                    let mut offset = 0;
+                    let response_buffer = response.data();
+                    let pub_key_len: usize = (response_buffer[offset] - 1).into();
+                    offset += 1;
```

Comment:
Could you explain this line?

Reply:
Yeah good question. Basically we want to increment the offset or index, because index 0 is where is pk length is. Then the actual PK starts at the next index.